client/logic/ui/game_input.gd

```diff
@@ -6,8 +6,8 @@ signal input_second
 signal input_escape
 signal selection_changed(terrain: Terrain)
 
-@onready var cursor: Cursor = $Cursor
-@onready var selection: Selection = $Selection
+@onready var cursor: Cursor = $Decouple/Cursor
```

Comment:
what are you trying with this decouple?

Reply:
Since they are children of the camera, their positions are transformed along with the camera's movement. Therefore, moving the camera causes selections to become misaligned. To address this issue, I decoupled them from the camera by introducing a normal node (Decouple) in between, breaking the connection between their position and that of the camera.